src/hint_processor/builtin_hint_processor/math_utils.rs

```diff
@@ -117,41 +117,49 @@ pub fn assert_le_felt(
         .ok_or_else(|| HintError::MissingConstant(Box::new(PRIME_OVER_2_HIGH)))?;
     let a = &get_integer_from_var_name("a", vm, ids_data, ap_tracking)?
         .clone()
-        .into_owned();
+        .into_owned()
+        .to_biguint();
     let b = &get_integer_from_var_name("b", vm, ids_data, ap_tracking)?
         .clone()
-        .into_owned();
+        .into_owned()
+        .to_biguint();
     let range_check_ptr = get_ptr_from_var_name("range_check_ptr", vm, ids_data, ap_tracking)?;
 
+    // TODO: use UnsignedInteger for this
+    let prime_div2 = prime_div_constant(2)?;
+    let prime_div3 = prime_div_constant(3)?;
+
     if a > b {
-        return Err(HintError::NonLeFelt252(Box::new((a.clone(), b.clone()))));
+        return Err(HintError::NonLeFelt252(Box::new((
+            Felt252::from(a.clone()),
+            Felt252::from(b.clone()),
```

Comment:
Do we need to `clone` here? Won't the reference do?

Reply:
We don't have a `From<&BigUint>` implementation. Although, I found a way to remove these: I was taking a reference to `a` and `b` unnecessarily, just removing that will do.